project/jasonchen1221/README.md

```diff
@@ -0,0 +1,82 @@
+
+
+## Problem to Solve
+For this project, I will try to implement the R-Tree search engine to meet the requirements as below,
+	1. It works in two-dimensional space and may index point, line, and surface.
+	2. Allow dynamic update of elements.
+	3. Allow access elements using a serial (integer) identifier.
+	4. Support ranged search of the geometrical entities.
+	5. Try to apply the R-Tree search engine on the EDA problems on hand.
+
+
+## Prospective Users
+The R-Tree search engine is for fast spatial searching. Electronic Design Automation(EDA) problems, map applications, and the problems which have large amount of data on the two-dimensional space with coordinates, shapes, and other information which related to the space may take advantage of it's fast speed of space searching.
+
+
+## System Architecture
+_(Still need to think about and work on this part as I think more about this project and know more about how to plan the system architecture…)_
+
+Basically, I will mainly use C++ to do the implementation of the engine for the reason of the high speed operation of R-Tree. And I will use high level language Python to test and verify the correct operation of the R-Tree search engine.
+
+To meet the requirements as mentioned in the section of Problem to Solve, the system architecture would base on the main class "RTREE". Also I may have some different functions to support the operations like dynamically adding a new object, deleting an object, and querying the information at all times.
+
+For the consideration of the problem size and the speed of these operation, I may need to consider which STL container to use and how to do the implementation to make sure the system is efficient enough. I will try to analyze the runtime and memory consumption of this R-Tree search engine. The ways to analyze are to be determined.
+
+
+## API Description
+_(Still need to think about and work on this part as I think more about this project and know more about how to consider different aspects about API…)_
+
+Public API will be wrapped to Python using pybind11 for testing the correctness of R-Tree search engine. e.g. Using Python to inserting nodes, querying the region, reading the file information…etc. And testing the correctness of R-Tree search engine.
+
+
+## Engineering Infrastructure
+In this project, 
+	1. Development : For the faster operation speed of R-Tree search engine, I will mainly use C++ to do the implementation of the engine.
+	2. Testing : Public API will be wrapped to Python using pybind11 for testing the correctness of R-Tree search engine.
+	3. Version Control : using Git.
+
+
+## Schedule
+**Before Week 1:**
+* Study the paper of R-Tree, and think more about the system architecture and implementation of R-Tree search engine.
```

Comment:
It is a good idea to study the R-Tree before coding.  It greatly help to write some snippets to prototype the code.